File: 15784.py
N, a, b = map(int, input().split())

seat = []
for i in range(N):
    students = list(map(int, input().split()))
    seat.append(students)

angry = False
jinseo = seat[a - 1][b - 1]
for i in range(N):
    if seat[a - 1][i] > jinseo:
        angry = True
    if seat[i][b - 1] > jinseo:
        angry = True

if angry: print("ANGRY")
else: print("HAPPY")

File: 15784.cpp
#include <iostream>
using namespace std;

int seat[1000][1000];

int main() {
	int N, a, b; cin >> N >> a >> b;

	for (int i = 0; i < N; i++)
		for (int j = 0; j < N; j++)
			cin >> seat[i][j];

	bool angry = false;
	int jinseo = seat[a - 1][b - 1];
	for (int i = 0; i < N; i++) {
		if (seat[a - 1][i] > jinseo)
			angry = true;
		if (seat[i][b - 1] > jinseo)
			angry = true;
	}

	if (angry) cout << "ANGRY\n";
	else cout << "HAPPY\n";
}
